src/Pulsar.Client/Internal/ConsumerImpl.fs

```diff
@@ -161,6 +163,28 @@ type ConsumerImpl internal (consumerConfig: ConsumerConfiguration, clientConfig:
         | Some startMsgId ->
             subscriptionMode = SubscriptionMode.NonDurable && startMsgId.LedgerId = msgId.LedgerId && startMsgId.EntryId = msgId.EntryId
 
+    let clearDeadLetters() =
+        deadLettersProcessor
+        |> Option.iter (fun p -> p.ClearMessages())
```

Comment:
Option.iter doesn't look good for me, can we implement DISABLED version to make the code cleaner?

Reply:
Done!